### PR TITLE
fix-square bracket : V1

### PR DESCRIPTION
**Pull request by LeComptoirDesPharmacies**
Fixes # . 

Fix square bracket in query params. Example : `orderBy[eq]`.

Its a quick fix to permit square bracket in query params, if they are define in the schema.

For query params who are not in the schema. Do default computation for deep object.

Changes proposed in this pull request:

 - Modify `sanitized` function to replace `[` by `_`. Example : `orderBy[eq] -> order_by_eq`
 - Modify `_make_deep_object` function to use key with square bracket in the name as a dict key if its define in the schema

### CODE REVIEW

**Comment by LeComptoirDesPharmacies** [comment]
Its resolving this issues : https://github.com/zalando/connexion/issues/1271

**Comment by RobbeSneyders** [comment]
Hi @LeComptoirDesPharmacies, thanks for the PR.
Could you add a test that shows the expected behavior?

**Comment by LeComptoirDesPharmacies** [comment]
Sorry for the delay, the test is ok now

**Comment by LeComptoirDesPharmacies** [comment]
Hi,

This PR is waiting for another review. Can someone do it please ?

Best regards,
Gery

File: tests/decorators/test_uri_parsing.py
Comment (RobbeSneyders) [inline tests/decorators/test_uri_parsing.py:12]: What is the query string that leads to this `MultiDict`?

File: tests/decorators/test_uri_parsing.py
Comment (RobbeSneyders) [inline tests/decorators/test_uri_parsing.py:114]: As far as I know, there is no dict collection format, is there?

**Review by RobbeSneyders — COMMENTED** [review]
Thanks Gery! I left some comments, could you clarify?

File: tests/decorators/test_uri_parsing.py
Comment (LeComptoirDesPharmacies) [reply tests/decorators/test_uri_parsing.py:12]: Its a API Rest design named LHS Brackets : 
- https://christiangiacomi.com/posts/rest-design-principles/

**Comment by LeComptoirDesPharmacies** [comment]
@RobbeSneyders, sorry for the delay, thanks for your review. 

I took into account your comments.

Best regards,
Gery

**Review by RobbeSneyders — APPROVED** [review]

**Comment by RobbeSneyders** [comment]
Thanks @LeComptoirDesPharmacies!

**Pull request merged**
